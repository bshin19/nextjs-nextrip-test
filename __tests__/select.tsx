import { fireEvent, render } from "@testing-library/react";
import Select from "../components/select";

describe("Select", () => {
  const required = {
    id: "foo",
    description: "bar",
  };

  it("should invoke onChange", () => {
    const testid = "dogs";
    const onChange = jest.fn();

    expect(onChange).not.toHaveBeenCalled();

    const { getByTestId } = render(
      <Select {...required} data-testid={testid} onChange={onChange}>
        <option>text</option>
      </Select>
    );

    fireEvent.change(getByTestId(testid));
    expect(onChange).toHaveBeenCalled();
  });

  it("should match minimum config", () => {
    const { container } = render(<Select {...required}></Select>);
    expect(container).toMatchSnapshot();
  });

  it("should match fully configured", () => {
    const { container } = render(
      <Select {...required} data-x="my_data" name="my_name">
        <option>floobar</option>
      </Select>
    );

    expect(container).toMatchSnapshot();
  });
});
