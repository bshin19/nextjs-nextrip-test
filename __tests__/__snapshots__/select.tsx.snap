// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should match fully configured 1`] = `
<div>
  <label
    class="mb-.5 bold text-center"
    for="foo"
  >
    bar
  </label>
  <select
    class="fs-1 mb-1"
    data-x="my_data"
    id="foo"
    name="my_name"
  >
    <option>
      floobar
    </option>
  </select>
</div>
`;

exports[`Select should match minimum config 1`] = `
<div>
  <label
    class="mb-.5 bold text-center"
    for="foo"
  >
    bar
  </label>
  <select
    class="fs-1 mb-1"
    id="foo"
    name="foo"
  />
</div>
`;
